tsnake/snake.py

```diff
@@ -5,45 +5,73 @@
 
 import numpy as np
 from scipy.linalg import solve_triangular
+from .utils import UtilPoint as uPoint
+from .utils import UtilEdge as uEdge
+from .utils import dist, seg_intersect
+# from tqdm import tqdm 
 
 # Implementation Notes: https://www.crisluengo.net/archives/217#more-217
 
-class Node(object):
+
+class Node(uPoint):
     """
     A class representing a single node in a T-snake.
     """
-    
+
     def __init__(self, x, y):
-        self.x = x  # todo: arg check if we care
-        self.y = y
-       
+        super().__init__(x, y)
+        self._normal = None
+    
+    def update(self, x, y):
```

Comment:
Isn't this also in the parent class uPoint? Do you need to override it here?

Reply:
Ah you're right it is I didn't see that, I'll drop the redundancy.